### ISSUE TITLE
search feature request - sorting results

### ISSUE THREAD

**Issue by cmichaelgraham** [gitterHQ/gitter#566]
seems kinda random to me how the results come back.  

it would be really nice if i could:
1. sort by person, then by date descending
2. sort by date descending

**Comment by PWKad** [comment]
To tack on to @cmichaelgraham here, it would be really nice to have a way to sort by your mentions so that when you get mentioned you can see when and where it was most recently, which is especially helpful in very busy chats.

**Comment by AshleyGrant** [comment]
Yeah, if search results came back in reverse chronological (sort by date descending) order by default, that would be awesome. The search feature is great, but having the seemingly random results ordering has made it rather less useful as the Aurelia channel has gotten bigger.

**Comment by gishmel** [comment]
I concur the reverse chronological (sort by date descending) seems to be a great default ordering method and +1 if you allow more customization on top of that. Currently it is as stated not as useful as it should be due to seemingly random return order.

**Comment by jdanyow** [comment]
:+1:

**Comment by mttmccb** [comment]
Please, this would be so useful, especially as older result can be irrelevant and thus the searching for a needle in the haystack starts.

**Comment by weaktyper** [comment]
:+1:

**Comment by ggranum** [comment]
+1 - The integrated search is nearly worthless without date ordering on active pages. Ok, ok, it's merely *excruciating*, not worthless.

**Comment by ejsmith** [comment]
Please!!!

**Comment by raine** [comment]
Please sort the results by time.

**Comment by mmcnl** [comment]
Any plans to implement this in the future?

**Comment by grahamperrin** [comment]
+1 
to allowing normal (not reverse) chronological order.

**Comment by samgoeta** [comment]
Dear Gitter, could you please offer reverse-chronological results? I am trying to add Gitter to [Zapier](www.zapier.com) but it's not possible without reverse-chronological data (see [documentation](https://zapier.com/developer/documentation/v2/triggers/#polling-url). 

Please fix it, Zapier will extend Gitter's features to hundreds of popular APIs such as Dropbox, Slack or Google Spreadsheet.

**Comment by grahamperrin** [comment]
Context: #1361 and please: the sooner, the better. 

(A recent use case: putting chronological order to the list of chat messages at https://discourse.trueos.org/t/what-is-wayland/284/4 was **horribly** time-consuming. It's not simply a case of drafting, and then manually sorting what's in the draft. It's the horror of ploughing through a disorderly list of search results, trying to make _timeline sense_ of things whilst the Gitter-given order fights against gaining that sense; whilst the targets pulse (expand and contract) so I must avoid pointing at them whilst dragging the …; and so on. The whole thing quickly becomes frustrating.)

A better UI to search results will allow Gitter to really shine. Thanks.

**Comment by AshleyGrant** [comment]
I was just told by Eric Eastwood that there are no plans to implement this utterly fundamental failing of Gitter. Now that Gitter is being open sourced, I've been told I'm free to implement search sorting myself.

This request has been ignored for two years now, and now we're being told to implement it ourselves. This is completely unacceptable. But I guess we have no choice but to accept it.

**Comment by grahamperrin** [comment]
@AshleyGrant if you'd like to chat about the transition to open source, first https://github.com/opensourcedesign/organization/issues/70 then (linked from the home page) http://chat.opensourcedesign.net/
